app/models/foreman_salt/concerns/hostgroup_extensions.rb

```diff
@@ -10,6 +10,9 @@ module HostgroupExtensions
         belongs_to :salt_proxy, :class_name => 'SmartProxy'
         belongs_to :salt_environment, :class_name => 'ForemanSalt::SaltEnvironment'
 
+        validates :salt_proxy, presence: true, unless: -> { salt_proxy_id.blank? }
```

Comment:
Best practice would be the `allow_blank` modifier at this point.
For some reason, the validation did not work with [allow_blank](https://guides.rubyonrails.org/active_record_validations.html#allow-blank) (or allow_nil) - it would just allow any input. This may be related to the fact that we want to check `salt_proxy_id`, but the `allow_blank` modifier allows a blank SmartProxy object in general.